android-network-speed/src/main/java/io/testproject/addon/android_network/AndroidNetworkAction.java

```
@@ -28,17 +28,18 @@
 
 @Action(name = "Set Network Speed",
         description = "Set the Android emulators perceived network speed",
-        summary = "Sets the synthetic network speed for an Android emulator to match one of a variety of pre-set network speeds")
+        summary = "Sets the synthetic network speed for an Android emulator to match " +
+                  "one of a variety of pre-set network speeds")
 public class AndroidNetworkAction implements AndroidAction {
 
-    @Parameter(description = "The network speed to adopt. Must be one of the following values: GSM, SCSD, GPRS, EDGE, UMTS, HSDPA, LTE, EVDO, FULL")
-    public String speed = "LTE";
+    @Parameter(description = "The network speed to adopt. Must be one of the following " +
+                             "values: GSM, SCSD, GPRS, EDGE, UMTS, HSDPA, LTE, EVDO, FULL")
+    public String speed = "";
 
     @Override
     public ExecutionResult execute(AndroidAddonHelper helper) throws FailureException {
         AndroidDriver<AndroidElement> driver = helper.getDriver();
         driver.setNetworkSpeed(NetworkSpeed.valueOf(speed));
```

What about checking the validity `speed` string value?
As far as I can tell `NetworkSpeed.valueOf()` throws an `IllegalArgumentException` when passed a string that doesn't match any of the enum values, and this exception is not handled...

---

I figured that this exception would be handled by the overall framework. Is there some other way to flag to the framework that we encountered an error?

---

You're probably right and the exception will be handled by the upper layer.
Was just wondering whether the exception message will be informative enough.
Let's wait for @mstrelex's input next week.

---

Sorry it took me only half year, but yes- FailureException is handled on the upper layer.